app/controllers/api/connections_controller.rb

```diff
@@ -153,4 +164,11 @@ def render_json_400(message)
         render(json: error, status: 400)
     end
 
+    private
+    def update_unread_notifications
+        # only informative notifications should be marked read
+        @unread_notifications.where(notification_type: ["match_accepted", "match_denied"]).update_all(read: true)
+
+    end
```

Comment:
In the method above, you return all notifications, but here you only mark some of them as read. Is there any use case where a notification with the types `received_request` or `connection_closed` would change `read` status?

Reply:
Yes! Those types of notifications should only be marked as read when the user interacts with them. a `received_request ` must be answered and a `connection_closed` enforces a review to be done. The others, as they are only visual, are marked automatiaclly as soon as they are received